File: Reflow/Provider/Text/TextLink.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace Reflow.Provider.Text
{
    public class TextLink : IDataLink
    {
        Dictionary<string, string> _connectionProperties = new Dictionary<string, string>();
        public bool Initialize(string connectionString)
        {
            string[] props = connectionString.Split(";".ToCharArray());
            foreach ( string prop in props)
            {
                string[] vals = prop.Split("=".ToCharArray());
                _connectionProperties.Add(vals[0], vals[1]);
            }
            return true;
        }

        public Dictionary<string, string> ConnectionProperties
        {
            get
            {
                return new Dictionary<string, string>(_connectionProperties);
            }
        }

        public bool IsConnected
        {
            get { return true; }
        }

        public bool Connect()
        {
            return true;
        }

        public bool Disconnect()
        {
            return true;
        }

        public ColumnDefinition[] GetSchema(string query)
        {
            TextReader reader = new TextReader();
            reader.Initialize(this);
            reader.Open();
            reader.Reader.Read();
            DataTable tbl = reader.Reader.GetSchemaTable();
            List<ColumnDefinition> items = new List<ColumnDefinition>();

            foreach (DataRow row in tbl.Rows)
            {
                ColumnDefinition col = new ColumnDefinition();
                col.ColumnName = row["ColumnName"] as string;
                col.Length = (int)row["ColumnSize"];
                col.DataType = DbType.String;
                col.IsLong = false;
                col.IsNullable = true;
                items.Add(col);
            }
            reader.Reader.Close();
            return items.ToArray();


        }

        public bool CreateTable(string tableName, ColumnDefinition[] columns, bool shouldDropExisting)
        {
            throw new NotImplementedException();
        }
    }
}

File: Java/src/main/java/com/nuarca/etl/provider/text/TextLink.java
package com.nuarca.etl.provider.text;


import CS2JNet.System.StringSupport;
import com.nuarca.etl.ColumnDefinition;
import com.nuarca.etl.provider.IDataLink;

import java.sql.JDBCType;
import java.sql.ResultSetMetaData;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class TextLink   implements IDataLink
{
    HashMap<String,String> _connectionProperties = new HashMap<String,String>();
    public boolean initialize(String connectionString)  // throws Exception
    {
        String[] props = connectionString.split(StringSupport.charAltsToRegex(";".toCharArray()));
        for (String prop : props)
        {
            String[] vals = prop.split(StringSupport.charAltsToRegex("=".toCharArray()));
            _connectionProperties.put(vals[0], vals[1]);
        }
        return true;
    }

    public HashMap<String,String> getConnectionProperties() throws Exception {
        return new HashMap<String,String>(_connectionProperties);
    }

    public boolean getIsConnected()   throws Exception
    {
        return true;
    }

    public boolean connect()  throws Exception
    {
        return true;
    }

    public boolean disconnect()   throws Exception
    {
        return true;
    }


    public ColumnDefinition[] getSchema(String query) throws Exception {

        //throw new Exception("Not implemented");
        TextReader reader = new TextReader();
        reader.initialize(this);
        reader.open();
        reader.getReader().next();
        ResultSetMetaData meta = reader.getReader().getMetaData();
        List<ColumnDefinition> items = new ArrayList<ColumnDefinition>();
        int colCount = meta.getColumnCount();
        for (int i=0; i<colCount;i++)
        {
            ColumnDefinition col = new ColumnDefinition();
            col.setColumnName(meta.getColumnName(i));
            col.setLength(meta.getPrecision(i));
            col.setDataType(JDBCType.VARCHAR);
            col.setIsLong(false);
            col.setIsNullable(true);
            items.add(col);
        }
        reader.getReader().close();
        return ((ColumnDefinition[]) items.toArray());
    }



    public boolean createTable(String tableName, ColumnDefinition[] columns, boolean shouldDropExisting) throws Exception
    {
        throw new Exception("Not Implemented");
    }

}

